### ISSUE TITLE
[REGRESSION]: UI mode restarts if keep storage state

### ISSUE THREAD

**Issue by dtokarczyk** [microsoft/playwright#30300]
### Version

1.43.0

### Steps to reproduce

1. Clone this repo `https://github.com/dtokarczyk/pw-storage-issue`
2. Run `npm install`
3. Run `npx playwright install`
4. Run `npm run test:ui`

### Expected behavior

I should see the results of my test (action steps, green checkmark etc)

### Actual behavior

After the end of the tests UI mode restarts and lost all things from UI mode. 

### Additional context

It just happens only in UI mode, in headless works fine. 

https://github.com/microsoft/playwright/assets/12712293/0effffd6-197f-4b13-9eb3-fbc6a017e220

### Environment

```shell
System:
    OS: macOS 14.3
    CPU: (11) arm64 Apple M3 Pro
    Memory: 750.98 MB / 36.00 GB
  Binaries:
    Node: 20.11.1 - /usr/local/bin/node
    Yarn: 1.22.21 - /opt/homebrew/bin/yarn
    npm: 10.2.4 - /usr/local/bin/npm
  Languages:
    Bash: 3.2.57 - /bin/bash
  npmPackages:
    @playwright/test: 1.43.0 => 1.43.0
```

**Comment by dil-koaru** [comment]
Same issue here!

**Comment by mxschmitt** [comment]
I'm able to repro!

**Comment by LillieEntur** [comment]
For others that have same issue.
I downgraded to 1.42.1 to get it working again.

**Comment by carlosrsabreu** [comment]
Same issue here!

**Comment by mxschmitt** [comment]
Reopening until released as a patch release. Will be available as `npm install @playwright/test@beta` in a bit.

**Comment by mxschmitt** [comment]
This should be fixed in v1.43.1.

**Comment by karlhorky** [comment]
@dtokarczyk @mxschmitt Thanks for reporting this and fixing the issue! 🙌

I can confirm that `@playwright/test@1.43.1` fixes the issue for us.

I reported a duplicate issue over here:

- https://github.com/microsoft/playwright/issues/30404

Adding some keywords here because I was unable to find this existing issue using the search:

- test results cleared
- test results disappear
- test results disappearing
- delay
- after test run
- Did not run
